demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -441,6 +1191,25 @@ static bool getS3ObjectFileSize( size_t * pFileSize,
     char * contentRangeValStr = NULL;
     size_t contentRangeValStrLength = 0;
 
+    /* Variable to store hash digest of payload. */
+    char * pPayloadHashDigest = ( char * ) malloc( SHA256_HASH_DIGEST_LENGTH );
+    /* Variable to store hex encoded hash digest of payload. */
+    char * hexencoded = ( char * ) malloc( SHA256_HEX_ENCODED_DIGEST_LENGTH );
+
+    /* Store Authorization header value generated using SigV4 library. */
+    char * pSigv4Auth = ( char * ) malloc( AWS_HTTP_AUTH_HEADER_VALUE_LEN );
+    size_t sigv4AuthLen = AWS_HTTP_AUTH_HEADER_VALUE_LEN;
```

Comment:
Same comment here about memory allocation either in the stack or in the global space statically